remove redundant code

JPrentice-thecode opened the pull request:
https://core.trac.wordpress.org/ticket/60052#no0

 that causes crit errors if default taxonomies catergory / post_tag have been removed or disabled.

The following line is a more generic check that would check category and post tag if they existed.

Deleted:	wp_delete_object_term_relationships( $post_id, array( 'category', 'post_tag' ) );

This following line does the same thing but doesn't fail if empty, and covers any custom taxonomies.

Following line:
	wp_delete_object_term_relationships( $post_id, get_object_taxonomies( $post->post_type ) );

JPrentice-thecode committed:
remove redundant code that causes crit errors if default taxonomies catergory / post_tag have been removed or disabled.

The following line is a more generic check that would check category and post tag if they existed.

danielbachhuber commented on src/wp-includes/post.php:
```diff
@@ -6369,7 +6369,6 @@ function wp_delete_attachment( $post_id, $force_delete = false ) {
 	do_action( 'delete_attachment', $post_id, $post );
 
-	wp_delete_object_term_relationships( $post_id, array( 'category', 'post_tag' ) );
 	wp_delete_object_term_relationships( $post_id, get_object_taxonomies( $post->post_type ) );
```
This line was originally added in https://core.trac.wordpress.org/changeset/12305

I'm hesitant to remove the first line without knowing why the second line was needed.

As a next step, I'd suggest writing unit tests covering this scenario that prove the behavior remains the same after the change.

I also think we need to figure out why both lines were included in the first place, and what changed since then that makes the first line no longer relevant.

danielbachhuber replied:
> I also think we need to figure out why both lines were included in the first place, and what changed since then that makes the first line no longer relevant.

As it turns out, the answer is in the original Trac ticket! https://core.trac.wordpress.org/ticket/10821#comment:7

We'll need to keep the line, but we can add some better safeguarding around it.

lenaseo22 commented:
That should work 👍
{{{
#!html
<a href="https://bytevarsity.ca/"></a><a href="http://ukvibesonline.co.uk/"></a><a href="https://dcanews.ca/"></a><a href="https://canadahomesnews.ca/"></a><a href="https://blesa.ca/"></a><a href="https://vixrim.ca/"></a><a href="https://totimescanews.ca/"></a><a href="http://canadianews.ca/"></a><a href="http://newsag.ca/"></a><a href="http://ottawas.ca/"></a><a href="http://canadavista.ca/"></a><a href="https://cryptowavezen.com/"></a><a href="https://bitcryptomining.net/"></a><a href="https://cryptocoinsupdates.com/"></a><a href="https://forexcryptonews.net/"></a><a href="https://bitforexvista.com/"></a><a href="https://stocktradalert.com/"></a><a href="https://tradingsentiment.org/"></a><a href="https://forexinvester.com/"></a><a href="https://stocksnotify.com/"></a><a href="https://stocktrendsalert.com/"></a><a href="https://topforexebroker.com/"></a><a href="https://forexah.com/"></a><a href="https://forexinsides.com/"></a><a href="https://stocksrush.com/"></a>
}}}

JPrentice-thecode committed:
Instead of removing the line, the line needs safety checks.

JPrentice-thecode committed:
simplify the check.

JPrentice-thecode committed:
going in different direction, dup bad. Needs new function that gets all taxonomies that have a term attached to the given attachment regardless if the the tax is attached.

JPrentice-thecode committed:
wrong unit test, passing

JPrentice-thecode committed:
Change for passing to dif branch

jopdop30 committed:
Remove duplication of wp_delete_object_term_relationships

JPrentice-thecode committed:
New changes, taxonomy has error checks and exists check.

anthonythorne committed:
Unit tests: Verify wp_delete_attachment() handles missing default taxonomies (#60052)

Adds a test that unregisters 'post_tag' before deleting an attachment, reproducing
the scenario in ticket #60052. Ensures wp_delete_object_term_relationships() and
wp_delete_attachment() no longer produce a fatal error when default taxonomies
are removed.

anthonythorne committed:
Merge pull request #1 from anthonythorne/bugfix/60052-remove-redundant-default-tax-delete-object-term-relationships-athorne

Bugfix/60052 remove redundant default tax delete object term relationships athorne

anthonythorne committed:
Merge branch 'bugfix/60052-remove-redundant-default-tax-delete-object-term-relationships' into bugfix/60052-remove-redundant-default-tax-delete-object-term-relationships-athorne

anthonythorne committed:
Merge pull request #3 from anthonythorne/bugfix/60052-remove-redundant-default-tax-delete-object-term-relationships-athorne

Bugfix/60052 remove redundant default tax delete object term relationships athorne

JPrentice-thecode committed:
$taxonomies not optional.

JPrentice-thecode committed:
remove unit test.

JPrentice-thecode committed:
Merge branch 'trunk' into bugfix/60052-remove-redundant-default-tax-delete-object-term-relationships

JPrentice-thecode committed:
phpcs

JPrentice-thecode committed:
phpcs

JPrentice-thecode committed:
phpcs

anthonythorne committed:
Fix unit test for attachment deletion when 'post_tag' is unregistered

Ensures the test properly unregisters 'category' and 'post_tag' before deleting an attachment.
Explicitly unsets `$wp_taxonomies` entries to fully simulate the missing taxonomy scenario.
This aligns with the issue in #60052, verifying that `wp_delete_attachment()` handles unregistered taxonomies gracefully.

JPrentice-thecode committed:
Merge pull request #4 from anthonythorne/bugfix/60052-remove-redundant-default-tax-delete-object-term-relationships-athorne

Fix unit test for attachment deletion when 'post_tag' is unregistered

anthonythorne committed:
remove unnecessary spacing

anthonythorne committed:
Merge pull request #5 from anthonythorne/bugfix/60052-remove-redundant-default-tax-delete-object-term-relationships-athorne

remove unnecessary spacing
